### DISCUSSION TITLE
Problems with chooser.py (Cannot find firmware)

### DISCUSSION THREAD

**Discussion by GideonLombardo** [USC-ACTLab/crazyswarm#707] [Q&A]
Hello,
I have recently encountered a new issue when I try to run:

python3 chooser.py

The error message returns:
```
WARNING: Could not find STM32 firmware (/home/workstation3/crazyswarm/ros_ws/src/crazyswarm/scripts/../../../../crazyflie-firmware/cf2.bin).
WARNING: Could not find NRF51 firmware (/home/workstation3/crazyswarm/ros_ws/src/crazyswarm/scripts/../../../../crazyflie2-nrf-firmware/cf2_nrf.bin).
Traceback (most recent call last):
  File "chooser.py", line 63, in <module>
    allCrazyflies = read_by_id(os.path.join(args.configpath, "allCrazyflies.yaml"))
  File "chooser.py", line 48, in read_by_id
    root = yaml.load(ymlfile)
TypeError: load() missing 1 required positional argument: 'Loader'
```
I replace the cradio.bin and chooser.py files to make sure they were not the issue, and I continue to get the same error message. This error had not happened previously, and I am not sure what changed. The only thing I have done is install a remote desktop on my machine so that I could run simulations while I was not in the lab, but I don't imagine this would have corrupted any files. 

Please let me know if there are any suggestions.

**Comment by GideonLombardo** [comment]
I actually was able to get things to work by specifying a loader with the yaml.load() command. However, I am still interested about the issue with my firmware, as that is a new WARNING message that I am getting.

**Comment by darshit-desai** [comment]
@GideonLombardo What loader did you specify? The solution doesn't seem clear I am also facing the same issue

**Comment by darshit-desai** [comment]
Never mind I figured it out, I need to replace yaml.load with yaml.safe_load at some locations
